### PR TITLE
feature(unlock-app): add ability to save Gitcoin passport verification hook for a lock

### LINKED ISSUE #13208: Gitcoin Passport hook
Can we create a Hook for `onKeyPurchase` that would require the recipient of the NFT to have a Gitcoin passport?

### PR DESCRIPTION
**Pull request by 0xTxbi**
# Description
This PR adds the ability for a lock manager to "select" the Gitcoin Passport Hook in the lock settings.

before – select gitcoin passport hook
select gitcoin passport hook
after – select gitcoin passport hook

# Issues
Fixes #13208
Refs #13208

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by 0xTxbi** [comment]
the CI fails due to type support for the Gitcoin Passport hook not yet introduced from unlock packages. Hook type support has been provided in this PR: #13435

**Comment by julien51** [comment]
Thanks! Can you please post screenshots as well?

**Comment by 0xTxbi** [comment]
definitely! updated the PR summary with the necessary screen grabs

**Review by julien51 — APPROVED** [review]

**Comment by 0xTxbi** [comment]
The UI has now been updated to accommodate custom minimum scores set by the lock manager.

improved UI

File: unlock-app/src/components/interface/locks/Settings/forms/hooksComponents/GitcoinContractHook.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        members with a required Gitcoin passport score.
+      </p>
+
+      <Input
+        size="small"
+        description="Required Gitcoin Passport Score"
+        type="number"
+        {...register('hook.requiredGitcoinPassportScore', {
+          valueAsNumber: true,
+          min: 0,
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/forms/hooksComponents/GitcoinContractHook.tsx:64]:
```suggestion
          min: 15,
```
I wanted to learn a bit more about the score and they have [this in the docs](https://support.gitcoin.co/gitcoin-knowledge-base/gitcoin-passport/common-questions/how-is-gitcoin-passports-score-calculated)

> The threshold for Gitcoin Passport Scoring is 15 and the maximum possible score is 100. Any score above 15 will be eligible for matching funding for their donations.

**Review by julien51 — APPROVED** [review]
LGTM!

File: unlock-app/src/components/interface/locks/Settings/forms/hooksComponents/GitcoinContractHook.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  useGetLockSettings,
+  useSaveLockSettings,
+} from '~/hooks/useLockSettings'
+
+export const GitcoinContractHook = ({
+  lockAddress,
+  network,
+  setEventsHooksMutation,
+}: CustomComponentProps) => {
+  const { isLoading: isLoadingSettings, data: settings } = useGetLockSettings({
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/forms/hooksComponents/GitcoinContractHook.tsx:15]:
```suggestion
  const { data: settings } = useGetLockSettings({
```

**Pull request merged**